"lpthw:ex48 unit test(using nosetests), classes,method arguments typeerrors hello i am newbie working through lpthw this is the link for it.at ex48 i am given a lot of code as unit tests to wich i am supposed to write the corresponding my_code wich will be tested(using nosetests, sry lots of tests in this sentence.)..this is my_code:     ..class lexicon(object):..    def __init__(self):.    #some initial stuff...    def scan(self,stuff):.    #some cool code....and this is the test.py file..from nose.tools import*.from ex48.code import lexicon..def test_directions():    ..    assert_equal(lexicon.scan(""north""), [('direction', 'north')]).    ...#unimportant details the above is enough....however when i run nosetests i get the following error: ..traceback (most recent call last):.    file ""c:blank31libsite-packagesnose'case.py"", line 197, i runtest.    ... line x, in test_directions.    assert_equal(lexicon.scan(""north""), [('direction', 'north')]).typeerror: scan() takes exactly 2 positional arguments (1 given)...wich is odd because one of those 2 arguments is the self one and i thought it should not be given...to remedy this problem i have tried:..@staticmethod.def scan(self,stuff):...but this gave me the same typeerror..next i tried:..assert_equal(lexicon.__init__(self).scan(...)...but this didn't work either(i think ._init_(self) should be used in another class than lexicon only, not in main)...i finally resolved to:..lexicon1=lexicon().def test_directions():    .    assert_equal(lexicon1.scan(""north""), [('direction', 'north')])...but this is not how the exercise is supposed to be done!.please help!sry about the long post!..p.s:i did not include the whole traceback bc it might not be necessary(and it's getting very late).if anybody thinks otherwise plz comment so i'll add it..i did not include parts of my code bc i'm sure that it works fine and is not causing the error!"
